kythe/cxx/extractor/bazel_artifact_selector.cc

```diff
@@ -214,19 +238,22 @@ void AspectArtifactSelector::ReadFilesInto(
     return;
   }
 
-  if (auto iter = state_.filesets.find(id); iter != state_.filesets.end()) {
+  if (auto node = state_.filesets.extract(id); !node.empty()) {
     state_.disposed.insert(std::string(id));
-    auto node = state_.filesets.extract(iter);
-    const build_event_stream::NamedSetOfFiles& filesets = node.mapped();
-
-    for (const auto& file : filesets.files()) {
-      files.push_back({
-          .local_path = AsLocalPath(file),
-          .uri = AsUri(file),
-      });
+    const FileSet& fileset = node.mapped();
```

Comment:
I'll just note that with everything named "file" or "iter" it's a little challenging to follow what's going on.